src/sentry/tasks/groupowner.py

```diff
@@ -20,49 +23,56 @@
 logger = logging.getLogger("tasks.groupowner")
 
 
-def process_suspect_commits(event, **kwargs):
+@instrumented_task(
+    name="sentry.tasks.process_suspect_commits",
+    queue="group_owners.process_suspect_commits",
+    default_retry_delay=5,
+    max_retries=5,
+)
+def process_suspect_commits(event_id, event_platform, event_frames, group_id, project_id, **kwargs):
     metrics.incr("sentry.tasks.process_suspect_commits.start")
-    with metrics.timer("sentry.tasks.process_suspect_commits"):
-        can_process = True
-        # Abbreviation for "workflow-owners-ingestion:group-{}"
-        cache_key = "w-o-i:g-{}".format(event.group_id)
+    lock = locks.get("w-o:{}-d-l".format(group_id), duration=10,)
+    try:
+        with lock.acquire(), metrics.timer("sentry.tasks.process_suspect_commits"):
+            # Abbreviation for "workflow-owners-ingestion:group-{}"
+            cache_key = "w-o-i:g-{}".format(group_id)
+            if cache.get(cache_key):
+                # Only process once per OWNER_CACHE_LIFE seconds.
+                metrics.incr(
+                    "sentry.tasks.process_suspect_commits.skipped",
+                    tags={"detail": "too_many_owners"},
+                )
+                return
```

Comment:
1/2: Since these are essentially all debouncing techniques, we should be applying them _before_ we schedule the task to drastically reduce the amount of message passing between tasks just to say "oh ok cool, debounced. nothing to do".